src/plugins/WebOfTrust/util/DelayedBackgroundJob.java

```diff
@@ -0,0 +1,26 @@
+package plugins.WebOfTrust.util;
+
+/**
+ * A {@link BackgroundJob} that is executed after triggers have been aggregated for some delay.
+ *
+ * @author bertm
+ */
+public interface DelayedBackgroundJob extends BackgroundJob {
+    /**
+     * Triggers the execution of the job after either aggregating triggers for at most some delay
+     * specified by the implementation or as soon as the currently running job finishes, whichever
+     * comes last. Implementations must ensure that this method is safe to invoke from any thread
+     * at any moment.
+     */
+    @Override
+    public void triggerExecution();
```

Comment:
There's a reason the definition is a bit mathematical: it appears slightly counterintuitive but actually is useful if you think about it. I tried describing it as a series of rules, but that quickly escalated into a set of overly verbose nested cases. If you don't mind it, I won't be changing this.